src/components/diagrams/singleLineDiagram/position-diagram.tsx

```diff
@@ -5,41 +5,55 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import React, { forwardRef, useEffect, useLayoutEffect, useRef, useState } from 'react';
-import PropTypes from 'prop-types';
-
+import React, { forwardRef, useEffect, useLayoutEffect, useRef, useState, Ref } from 'react';
 import { useSelector } from 'react-redux';
 
 import Box from '@mui/material/Box';
 import { useTheme } from '@mui/material/styles';
 import LinearProgress from '@mui/material/LinearProgress';
 import { SingleLineDiagramViewer } from '@powsybl/diagram-viewer';
-import { styles, MAX_HEIGHT_VOLTAGE_LEVEL, MAX_WIDTH_VOLTAGE_LEVEL, NoSvg, MIN_WIDTH } from '../diagram-common';
+import { styles, MAX_HEIGHT_VOLTAGE_LEVEL, MAX_WIDTH_VOLTAGE_LEVEL, NoSvg, MIN_WIDTH, Svg } from '../diagram-common';
 import { useIntlRef, useSnackMessage } from '@gridsuite/commons-ui';
 import { Paper } from '@mui/material';
 import DiagramHeader from '../diagram-header';
 import { fetchSvg } from '../../../services/study';
 import { mergeSx } from '../../utils/functions';
-
-const PositionDiagram = forwardRef((props, ref) => {
-    const [svg, setSvg] = useState(NoSvg);
+import { AppState } from 'redux/reducer';
+
+interface PositionDiagramProps {
+    diagramTitle: string;
+    svgUrl: string | null;
+    onClose: () => void;
+    svgType: string;
+    disabled?: boolean;
+}
+
+const PositionDiagram = forwardRef((props: PositionDiagramProps, ref: Ref<HTMLDivElement>) => {
+    // fixme : HTMLDivElement is the correct type ?
+    const [svg, setSvg] = useState<Svg>({
+        svg: null,
+        metadata: null,
+        additionalMetadata: null,
+        error: null,
+        svgUrl: null,
+    });
     const svgUrl = useRef('');
-    const svgDraw = useRef();
+    const svgDraw = useRef<SingleLineDiagramViewer | null>(null);
     const { snackError } = useSnackMessage();
     const intlRef = useIntlRef();
-    const svgRef = useRef();
+    const svgRef = useRef<HTMLDivElement>();
     const { svgType, disabled } = props;
 
-    const currentNode = useSelector((state) => state.currentTreeNode);
+    const currentNode = useSelector((state: AppState) => state.currentTreeNode);
 
-    const [loadingState, updateLoadingState] = useState(false);
+    const [loadingState, updateLoadingState] = useState<boolean>(false);
 
     const theme = useTheme();
 
     // using many useState() calls with literal values only to
     // easily avoid recomputing stuff when updating with the same values
-    const [serverHeight, setServerHeight] = useState(0);
-    const [serverWidth, setServerWidth] = useState(0);
+    const [serverHeight, setServerHeight] = useState<number>(0);
```

Comment:
Note => remove type typescript infer it with default value

Reply:
done